### ISSUE TITLE
Add JNI bindings for macos M1 (darwin-aarch64) platforms

### ISSUE THREAD

**Issue by sstone** [ACINQ/secp256k1-kmp#54]

**Comment by sstone** [comment]
The underlying issue is that github CI does not yet support M1 VMs (see https://github.com/actions/runner-images/issues/2187).

I think there may be a simple but ugly workaround though: M1 macs are supposed to have an emulation feature to run x86_64 code, so maybe extracting the x86 library and using it (with `-Dfr.acinq.secp256k1.lib.path`) would work ? (performance would not be very good I guess). I'll give it a try and update https://github.com/ACINQ/secp256k1-kmp/issues/54, and if it works I'll duplicate the JNI lib in the JAR so it will work out of the box.

**Comment by CristianMG** [comment]
Hi, could you develop an explanation of the workaround, please?

Added library for x86_64

`    runtimeOnly("fr.acinq.secp256k1:secp256k1-kmp-jni-jvm-darwin:${BuildConstants.secp_256_k1}")
`

Added JVM option to IntelliJ testing task.
-Dfr.acinq.secp256k1.lib.path=../../native/darwin-x86_64/libsecp256k1-jni.dylib

The option is not changing the path where java is searching the library

`Caused by: java.lang.IllegalStateException: No native library found: at /fr/acinq/secp256k1/jni/native/darwin-aarch64/libsecp256k1-jni.dylib`

Option executed finally

` :kostr-core:test --debug -Dfr.acinq.secp256k1.lib.path=../../native/darwin-x86_64/libsecp256k1-jni.dylib`

**Comment by sstone** [comment]
You would need to extract the x86_64 dynamic library from the `-jni-jvm-darwin` jar to a local directory and use it through `-Dfr.acinq.secp256k1.lib.path`. I think M1 macs are supposed to be able to run x86_64 libraries so it should work but I have not tried.

**Comment by kirillzh** [comment]
@sstone, wanted to pump this issue and check in to see if any chance that this could be prioritized? Would love to use `[secp256k1-kmp](https://github.com/ACINQ/secp256k1-kmp)` but blocked on lack of M1 support.

**Comment by sstone** [comment]
Not until  M1/M2 macos is supported by Github Actions which should happen this year (it's now on their roadmap). It's not a priority for us because M1/M2 macos is not a production target (as opposed to linux arm64 or ios arm64 for example). But in the meantime you can try @rorp's fork of this library (see https://github.com/ACINQ/eclair/issues/2427#issuecomment-1407148829).

**Comment by sstone** [comment]
@kirillzh @rorp can you please check if version 0.9.0-SNAPSHOT published on https://oss.sonatype.org/content/repositories/snapshots/ works for you ? it includes bindings for linux aarch64 and darwin aarch64

**Comment by rorp** [comment]
@sstone darwin aarch64 binding works like a charm for me. Thanks!
